### PR TITLE
Suggestions for service account script

### PR DESCRIPTION
**Pull request by chelnak**

### CODE REVIEW

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 [hunk 1]
```diff
@@ -34,10 +25,7 @@ The name of the Keyvault for the Environment
 $New-SqlDBAccountParameters = @{
 	 ServerName = ServerName
      DataBaseName = DataBaseName
-     SqlUserName = SqlUserName
-	 SqlPassword = SqlPassword
      SqlServiceAccountName = SqlServiceAccountName
-     SQLResourceGroup = SQLResourceGroup
      Enviroment = Enviroment
```
Comment (EwanNoble) [inline infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Spelling

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 [hunk 2]
```diff
@@ -46,31 +34,24 @@ $New-SqlDBAccountParameters = @{
...
+    [String]$AzureFirewallName = "AzureWebAppFirewall",
     [Parameter(Mandatory = $true)]
-    [String]  $SqlPassword,
+    [String]$DataBaseName,
     [Parameter(Mandatory = $true)]
-    [String]  $SqlServiceAccountName,
+    [String]$SqlServiceAccountName,
     [Parameter(Mandatory = $true)]
-    [String]  $SQLResourceGroup,
+    [String]$Enviroment,
```
Comment (EwanNoble) [inline infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Spelling

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 [hunk 3]
```diff
@@ -11,18 +11,9 @@ Name of the SQL Server
 .PARAMETER AzureFireWallName
```
Comment (EwanNoble) [inline infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Should probably use the new Az module throughout the script

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 [hunk 4]
```diff
@@ -11,18 +11,9 @@ Name of the SQL Server
...
 Name of the tempoary Sql Server FireWall rule created (Optional)
```
Comment (EwanNoble) [inline infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Spelling

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 [hunk 5]
```diff
@@ -46,31 +34,24 @@ $New-SqlDBAccountParameters = @{
...
-    [Parameter(Mandatory = $true)]
-    [String] $KeyVaultName
+    [String]$KeyVaultName
 )
 
 $ErrorActionPreference = 'Stop'
 
-
 function New-AzureSQLServerFirewallRule {
     $AgentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
```
Comment (EwanNoble) [inline infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Casing

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 [hunk 6]
```diff
@@ -79,79 +60,85 @@ function Update-AzureSQLServerFirewallRule {
...
-    $SecretName = "${Enviroment}-${SqlServiceAccountName}"
 
-    Write-Host "$SecretName"
+    $ServiceAccountSecretName = "$Enviroment-$SqlServiceAccountName".ToLower()
+    $ServerFQDN = "$ServerName.database.windows.net"
 
-    if ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue) -eq $null) {
+    # --- Retrieve SQL Server details
+    Write-Host "Searching for server resource $($ServerName)"
+    $ServerResource = Get-AzureRmResource -Name $ServerName -ResourceType "Microsoft.Sql/servers"
```
Comment (EwanNoble) [inline infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Look at README guidelines for casing rules on local variables

Comment (chelnak) [reply infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Don't fully agree with that - we should discuss @dalemorson @EwanNoble

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 (hunk 5 above)
Comment (chelnak) [reply infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Changed to:

```
(Invoke-WebRequest -Uri http://checkip.dyndns.com -UseBasicParsing).Content -Replace "[^\d\.]"
```

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 (hunk 6 above)
Comment (chelnak) [reply infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Ref #7

**Comment by EwanNoble** [comment]
Should use the Az module instead of AzureRm

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 [hunk 7]
```diff
@@ -32,126 +23,122 @@ The name of the Keyvault for the Environment
...
 function Get-RandomPassword {
     $Password = ([char[]]([char]33..[char]95) + ([char[]]([char]97..[char]126)) + 0..9 | Sort-Object { Get-Random })[0..33] -join ''
     return $Password
 }
 
-function Remove-AzureSQLServerFirewallRule {
-    if ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue)) {
-        Remove-AzureRmSqlServerFirewallRule -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+try {
+	$AgentIP = (Invoke-WebRequest -Uri http://checkip.dyndns.com -UseBasicParsing).Content -Replace "[^\d\.]"
```
Comment (EwanNoble) [inline infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Have previously used the below to find outbound IP, looks a little neater. Same outcome either way though.
```suggestion
	$AgentIP = (Invoke-WebRequest ifconfig.me/ip -UseBasicParsing).Content.Trim()
```

File: infrastructure-scripts/New-SqlDbServiceAccount.ps1 [hunk 8]
```diff
@@ -32,126 +23,122 @@ The name of the Keyvault for the Environment
...
-function New-AzureSQLServerFirewallRule {
-    $AgentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
-    New-AzureRmSqlServerFirewallRule -StartIPAddress $AgentIp -EndIPAddress $AgentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
-}
-function Update-AzureSQLServerFirewallRule {
-    $AgentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
-    Set-AzureRmSqlServerFirewallRule -StartIPAddress $AgentIp -EndIPAddress $AgentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
-}
-
 function Get-RandomPassword {
```
Comment (EwanNoble) [inline infrastructure-scripts/New-SqlDbServiceAccount.ps1]: Should this function be pulled out and shared with other scripts?

**Pull request merged**